ROMFS/px4fmu_common/init.d/rcS

```diff
@@ -217,25 +217,30 @@ else
 	fi
 	unset BOARD_RC_DEFAULTS
 
-	#
-	# Set parameters and env variables for selected SYS_AUTOSTART.
-	#
-	set AUTOSTART_PATH etc/init.d/rc.autostart
 	if ! param compare SYS_AUTOSTART 0
 	then
-		if param greater SYS_AUTOSTART 1000000
+		# rc.autostart directly run the right airframe script which sets the VEHICLE_TYPE
+		# Look for airframe in ROMFS
+		. ${R}etc/init.d/rc.autostart
+
+		if [ ${VEHICLE_TYPE} == none ]
 		then
-			# Use external startup file
+			# Look for airframe on SD card
 			if [ $SDCARD_AVAILABLE = yes ]
 			then
-				set AUTOSTART_PATH etc/init.d/rc.autostart_ext
+				. ${R}etc/init.d/rc.autostart_ext
 			else
-				echo "ERROR [init] SD card not mounted - trying to load airframe from ROMFS"
+				echo "ERROR [init] SD card not mounted - can't load external airframe"
 			fi
 		fi
-		. ${R}$AUTOSTART_PATH
+
+		if [ ${VEHICLE_TYPE} == none ]
+		then
+			echo "ERROR [init] No airframe file found for SYS_AUTOSTART value"
+			param set SYS_AUTOSTART 0
```

Comment:
Does resetting SYS_AUTOSTART to 0 actually accomplish anything or should we just leave it alone to keep things simple (development workflows, avoiding unnecessary resets jumping between versions, etc)?

Reply:
Resetting it to zero flags that the autostart wasn't loaded. When I talked to @bkueng he was stressing that this should be retained for the case someone updates but the airframe he used is gone from the new version. I remember I was once glad that I saw the ID o to zero in one of my tests. Not sure if that couldn't be solved even more user friendly e.g. by messageing the user that there's not airframe with the ID but not changing it.

For reference this was introduced here: https://github.com/PX4/PX4-Autopilot/pull/13772/files#diff-c0c83e328e28cbd2f7c7ad46bbb6369024ed63daa3f9dc8cf89ac79125183e1dR82
I did not change it but am open to discussion. I feel it's more accessible to change now.